docs/api-reference/index.md

```diff
@@ -0,0 +1,35 @@
+# API Reference
+
+## Classes
+
+```{eval-rst}
+.. currentmodule:: esssans
+
+.. autosummary::
+   :toctree: ../generated/classes
+   :template: class-template.rst
+   :recursive:
+```
+
+## Top-level functions and attributes
+
+```{eval-rst}
+.. autosummary::
+   :toctree: ../generated/functions
+   :recursive:
+
+   providers
+```
+
+## Submodules
+
+```{eval-rst}
+.. autosummary::
+   :toctree: ../generated/modules
+   :template: module-template.rst
+   :recursive:
+
+   beam_center_finder
```

Comment:
Is there a way we can improve the long function names? Maybe only have the function name and not the submodule path? But I guess that path is important information to have...
![Screenshot at 2023-10-04 15-16-28](https://github.com/scipp/esssans/assets/39047984/ecc5b16d-057c-41cd-b335-ed0943e43ef1)

Reply:
I suppose we can shorten, `beam_center_finder` -> `beam_center`, or remove the duplicate `beam_center_` prefix from the 2 finders?

Reply:
> remove the duplicate beam_center_ prefix from the 2 finders?

would work in this case. But I guess it won't save us from future functions with long names.

Reply:
Maybe having shorter module names is good style generally. Should we just rename `beam_center_finder` -> `beam`?

After all, PEP8 says: "Modules should have short, all-lowercase names. Underscores can be used in the module name if it improves readability."

Reply:
Or `beamcenter`? I feel `beam` is a little too vague?

Reply:
I suggest we keep it for now. We should probably sit down with Wojciech to review all the naming in the repo.